torch/csrc/distributed/c10d/ProcessGroupNCCL.hpp

```diff
@@ -181,6 +182,12 @@ static std::vector<std::string> TORCH_NCCL_USE_TENSOR_REGISTER_ALLOCATOR_HOOK =
     {"TORCH_NCCL_USE_TENSOR_REGISTER_ALLOCATOR_HOOK",
      "NCCL_USE_TENSOR_REGISTER_ALLOCATOR_HOOK"};
 
+// Return the rank of root and index during NCCL scalable comm init.
+static std::pair<int, int> getRootRankAndIndex(
+    const int rank,
+    const int nRanks,
+    const int nIds);
+
```

Comment:
No need to put this function in header file.

Reply:
oh you mean directly define in cpp? OK.